python/cudf/cudf/core/frame.py

```diff
@@ -1448,7 +1448,7 @@ def searchsorted(
 
         # Return result as cupy array if the values is non-scalar
         # If values is scalar, result is expected to be scalar.
-        result = cupy.asarray(outcol.data_array_view)
+        result = cupy.asarray(outcol.data_array_view(mode="read"))
```

Comment:
This should acquire a spill lock, I think, but it doesn't matter because `outcol` is new and doesn't escape this function.

Reply:
That's correct